async.md

```diff
@@ -0,0 +1,108 @@
+---
+---
+
+# A note on asynchronous code
+
+You might have noticed something in the small test we wrote in the introduction, namely the call to `await`. The Node.js engine is asynchronous, so when a call is made that takes some time to return (such as an API or Database call), rather than wasting time waiting for a response, the engine just processes the code that follows. This can result in issues when you try to use a variable that has not yet been initialized. For example, if you had:
+
+```
+test('hello!', () => {
+  response = await request(app).get('/')
+  expect(response.status).toBe(200);
+  expect(response.text).toEqual("Hello dxw!");
+});
+```
+
+`response.status` would return `undefined`, as the Node engine has not got round to initialising that variable. Traditionally, we've used callbacks to get around this, which would probably look something like this:
```

Comment:
If my comment above is right, then it's `response` itself that will be `undefined` here, so any property access on it will not work.

Reply:
👍